acorn/src/node.ts

```diff
@@ -8,6 +8,9 @@ export class Node {
   loc?: SourceLocation
   sourceFile?: string
   range?: [number, number]
+  body: any
+  sourceType: any
```

Comment:
Why are these two node-type-specific properties (for the `Program` node) declared here, but not others?

Reply:
Without them, `Parser#parseTopLevel: (node: Node) => Node` will fail [here](https://github.com/ninofiliu/acorn/blob/dcc251b027b762f7f345a16fc71927f3c4a5fb8a/acorn/src/statement.ts#L20) and in other places too

But maybe the typing of `parseTopLevel` is wrong and we don't need `.body` and `.sourceType`

Reply:
It's because `Node` is only a generic interface, whereas individual methods return specific node types.